vaadin-testbench-core/src/main/java/com/vaadin/testbench/unit/NotificationWrap.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2022 Vaadin Ltd
+ *
+ * This program is available under Commercial Vaadin Developer License
+ * 4.0 (CVDLv4).
+ *
+ *
+ * For the full License, see <https://vaadin.com/license/cvdl-4.0>.
+ */
+package com.vaadin.testbench.unit;
+
+import com.vaadin.flow.component.notification.Notification;
+import com.vaadin.flow.dom.DomEvent;
+import com.vaadin.flow.dom.Element;
+import com.vaadin.flow.internal.nodefeature.ElementListenerMap;
+
+import elemental.json.Json;
+
+/**
+ * Test wrapper for Notification components.
+ *
+ * @param <T>
+ *            component type
+ */
+public class NotificationWrap<T extends Notification> extends ComponentWrap<T> {
+
+    /**
+     * Wrap given component for testing.
+     *
+     * @param component
+     *            target component
+     */
+    public NotificationWrap(T component) {
+        super(component);
+    }
+
+    /**
+     * Gets the text of the notification.
+     *
+     * If the notification is not displayed an IllegalStateException will be
+     * thrown as the end user would not be able to see the contents.
+     * 
+     * @return the text of the notification
+     * @throws IllegalStateException
+     *             is notification is not displayed
+     */
+    public String getText() {
+        ensureComponentIsUsable();
```

Comment:
Actually should the notification isUsable only be about isAttached and isOpened?

Reply:
It makes sense to consider only isAttached and isOpened.
User can interact with Notification only if it contains interactive elements, and the tests should do the same.
It the notification is for some reason disabled, also children components will be, and their wrapper will prevent interactions.
Furthermore, also `close` method in wrapper may be removed, since there is no way to close a notification other than waiting for auto-close or having some component or shortcut that invokes `Notification.close`

Reply:
Would add also `isVisible` to the check